libcaf_core/caf/typed_response_promise.hpp

```diff
@@ -77,11 +100,43 @@ class typed_response_promise {
     return *this;
   }
 
+  /// Satisfies the promise by sending an empty message if this promise has a
+  /// valid message ID, i.e., `async() == false`.
+  void deliver(unit_t x) {
+    promise_.deliver(x);
+  }
+
```

Comment:
```suggestion
```

An actor that promises a tuple of type `Ts...` must not send an empty message instead. For "void" promises, `typed_actor_promise<unit_t>` should do the trick IIRC.